README.md

```diff
@@ -43,27 +43,17 @@ Create the environment variable settings needed to get Via 3 working nicely with
 
     make devdata
 
-### Run the services with Docker Compose
-
-Start the services that Via 3 requires (currently just NGINX) using Docker
-Compose:
-
-    make services
-
 ### Start the development server
 
     make dev
 
 The first time you run `make dev` it might take a while to start because it'll
 need to install the application dependencies and build the assets.
 
-This will start the server on port 9082 (http://localhost:9082), reload the
+This will start the server on port 9083 (http://localhost:9083), reload the
 application whenever changes are made to the source code, and restart it should
 it crash for some reason.
 
-To simulate how the app works in production, you should access it through NGINX
-on port 9093 (http://localhost:9083)
-
```

Comment:
I think it's right to front-and-center getting to this by NGINX, but we should still describe how the app works. It's not an internal detail and a new developer would need to understand this.

So I think I'd prefer this to say something along the lines of:

```
This will start the web server (http://localhost:9083), which is how
you should access the app underneath (http://localhost:9092) to replicate live. 

The application will reload whenever changes are made to the source code, 
and restart it should it crash for some reason.
```

Or you know... not terrible. 

Also, does this really restart if the app crashes? I thought the
benefit of `honcho` over `supervisor` is it didn't do this? I tried it by just raising in the 
app init, and it doesn't recover.

Reply:
There are separate sections further down in this README titled _How Via 3 works in production_ and _How Via 3 works in development_, so I think we should let those explain it, and let this _Installing..._ section stick to just giving quick and easy install instructions.

It doesn't restart the app on crash, no. That's weird, I'm not sure why that's in all the READMEs. We could correct that but on the other hand if we're going to move all the apps to supervisor then maybe it will be restarting them on crash soon enough

Reply:
Each one of our apps (h, lms, bouncer, via and via3) has a README that follows the same format. The section headings in the READMEs are:

1. Installing the app in a development environment

   This section is just quick-as-possible docs for telling the user what the commands to get the app going are. No diversions into explaining things about the app or how it works. Very consistent wording and formatting is used across all the READMEs. The subsections are always:

   1. You will need
   2. Clone the git repo (`git clone`)
   3. Run the services (`make services`)
   4. Create the development data and settings (`make devdata`)
   5. Start the development server (`make dev`)
   6. Run the tests (`make test`)

2. Further, app-specific discussion.

   Here, at the bottom of the file, is where each app's README can divert into app-specific discussion and link to further docs.

   For example LMS uses this section to explain SSL issues with LMS's and localhost development. Via has stuff about running with SSL in dev and about updating PDF.js. Bouncer has some random stuff here too. And this is where Via 3 has info about updating PDF.js, and about Via 3's architecture / how it works, including NGINX and Gunicorn, since Via 3 is somewhat unique in this respect.

The frontend apps client and browser-extension differ aggressively from this and from each other, but the five backend apps at least have very consistent READMEs.

Reply:
We can include the link later on in two pages text, and if someone misses it we can accurately say it's in the README. But I think in a larger way we've probably failed.

The following seems likely: A person follows the README step by step. Gets to the `make dev` part. The only URL you see in the output when starting up is `http://localhost:9082`. There's only one URL in the docs which is very similar. "That's the guy" they think and click on it, it mostly works, and then they get a CORS issue. They assume they've done something wrong and go _back_ over the steps they've already done (not reading ahead). End up confused.

A section I'd like to see at the top of our READMEs would look like this:

```
# Quick start

    git clone https://github.com/hypothesis/via3.git
    cd via3
    make devdata
    make dev

You should now have the following running:

 * http://localhost:9083 - Main app entry-point (NGINX docker image `via3_nginx-proxy_1`)
 * http://localhost:9082 - Pyramid app underneath
```

This would be particularly helpful in more complicated apps like `h`. 

This summarises all of the important info you need to know in one very small section you can copy paste, including hints at the architecture of the app (at least in dev). It also gives you the info you need to check if it's all working. 

We could then keep the existing sections which detail what `make devdata` is etc. and the far more detailed app explanation. The idea being each section adds detail, rather than much totally new info. If you stop reading sooner (and most do) you know a low-fi version of what you need to know. Think 'advertising' rather than DRY.

Reply:
Ah yes, I didn't think about it printing out 9082 to the terminal. Ok, I've added info about NGINX vs Gunicorn directly after the `make dev` command: https://github.com/hypothesis/via3/pull/195/commits/b43141d10681a43d2d43d7d87e9e6af34e529b7c

Reply:
Regarding the `# Quick start` idea: I think we _could_ add a quickstart section like that to the top of each app's README. It would have to mention that the prerequisites need to be installed first (otherwise the quickstart commands won't work). It would mean that the sequence of install commands is now duplicated in each app's README. Also, to be honest, although it's not as ultra-concise the **Installing `<APP>` in a development environment** section that each README currently has is intended to be a concise quick-start with a minimum of extraneous explanation and it already fits in roughly one screenful or less, so I think the benefit of adding an even-quicker-start section to the top would be minimal. I think these READMEs are pretty quick and easy to follow as-is.

All the apps have this, [including h](https://h.readthedocs.io/en/latest/developing/install/). We don't have more and less complicated install and setup among our apps -- they're all exactly the same.